src/components/MAPI/apps/AppList/AppDetail.tsx

```diff
@@ -0,0 +1,216 @@
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
+import { useHttpClientFactory } from 'lib/hooks/useHttpClientFactory';
+import RoutePath from 'lib/routePath';
+import { compare } from 'lib/semver';
+import AppInstallModal from 'MAPI/apps/AppInstallModal';
+import { extractErrorMessage } from 'MAPI/organizations/utils';
+import { GenericResponse } from 'model/clients/GenericResponse';
+import * as applicationv1alpha1 from 'model/services/mapi/applicationv1alpha1';
+import React, { useEffect, useMemo, useRef } from 'react';
+import { Breadcrumb } from 'react-breadcrumbs';
+import { useSelector } from 'react-redux';
+import { useDispatch } from 'react-redux';
+import { useRouteMatch } from 'react-router-dom';
+import { AppsRoutes } from 'shared/constants/routes';
+import DocumentTitle from 'shared/DocumentTitle';
+import { IState } from 'stores/state';
+import useSWR from 'swr';
+import { IVersion } from 'UI/Controls/VersionPicker/VersionPickerUtils';
+import AppDetailPage from 'UI/Display/Apps/AppDetailNew/AppDetailPage';
+
+import { isTestRelease } from '../utils';
+import {
+  fetchAppCatalogIndexAppVersionReadme,
+  fetchAppCatalogIndexAppVersionReadmeKey,
+  getAppCatalogIndexAppVersionReadmeURL,
+  getAppCatalogsIndexAppList,
+  getAppCatalogsIndexAppListKey,
+  IAppCatalogIndexAppList,
+  IAppCatalogIndexAppVersion,
+} from './utils';
+
+function mapAppCatalogIndexAppVersionsToReleasePickerItems(
+  appVersions?: IAppCatalogIndexAppVersion[]
+): IVersion[] | undefined {
+  if (!appVersions) return undefined;
+
+  return appVersions
+    .map((e) => ({
+      chartVersion: e.version,
+      created: e.created,
+      includesVersion: e.appVersion,
+      test: isTestRelease(e.version),
+    }))
+    .sort((a, b) => compare(b.chartVersion, a.chartVersion));
+}
+
+const AppDetail: React.FC<{}> = () => {
+  const match = useRouteMatch<{
+    catalogName: string;
+    app: string;
+    version: string;
+  }>();
+
+  const clientFactory = useHttpClientFactory();
+  const auth = useAuthProvider();
+
+  const appCatalogClient = useRef(clientFactory());
+  const { data: appCatalog, error: appCatalogError } = useSWR<
+    applicationv1alpha1.IAppCatalog,
+    GenericResponse
+  >(applicationv1alpha1.getAppCatalogKey('', match.params.catalogName), () =>
+    applicationv1alpha1.getAppCatalog(
+      appCatalogClient.current,
+      auth,
+      '',
+      match.params.catalogName
+    )
+  );
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (appCatalogError) {
+      const errorMessage = extractErrorMessage(appCatalogError);
+
+      new FlashMessage(
+        'There was a problem loading the app catalog.',
+        messageType.ERROR,
+        messageTTL.FOREVER,
+        errorMessage
+      );
+
+      ErrorReporter.getInstance().notify(appCatalogError);
+
+      dispatch(push(AppsRoutes.Home));
+    }
+  }, [appCatalogError, dispatch]);
+
+  const appCatalogList = appCatalog ? [appCatalog] : undefined;
```

Comment:
A bit ugly, but it allows me to re-use the same functions as in the `AppList` component